### PR TITLE
Updated dependencies, added functionality, ...

### PR DESCRIPTION
**Pull request by i997jn7**
- Updated dependencies
- Removed title so it can be set with browserWindowOverrides
- Added filepicker
- Directory button and dropdown default i18n ready
- Add button thats sends on ipc channel
- Add childBrowserWindowOverrides
- Fix cut off scrollbar
- Add ability to use own icons or none at all in sidebar

### CODE REVIEW

**Comment by jan-zajic** [comment]
@tkambler what about this PR? I will look forward to see file field get merged..

**Comment by pvrobays** [comment]
@i997jn7 I don't know how, but I must have overlooked this PR for quite a while now. I've been starting to help maintaining the repo since the start of 2021.
Meanwhile a few of your initial improvements have been implemented, but still quite some that aren't.

If you still want these other improvements to be merged, you can definitely create a new PR which I'd be glad to review and merge. However I think this one has deviated a bit too much.

@jan-zajic FYI, the file selector has meanwhile been implemented and released.

File: src/app/index.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import _ from 'lodash';
 
 const { ipcRenderer } = window.require('electron');
 const options = ipcRenderer.sendSync('getPreferenceOptions');
-const preferences = ipcRenderer.sendSync('getPreferences');
```
Comment (lacymorrow) [inline src/app/index.js:11]: ?

File: webpack.config.js [hunk 2]
```diff
@@ -4,6 +4,10 @@ const webpack = require('webpack');
 const path = require('path');
 
 module.exports = {
+    'target': 'node',
```
Comment (lacymorrow) [inline webpack.config.js:7]: why?

File: example/renderer.js [hunk 3]
```diff
@@ -13,3 +13,12 @@ ipcRenderer.on('preferencesUpdated', (e, preferences) => {
     console.log('Preferences were updated', preferences);
     prefsEl.innerHTML = JSON.stringify(preferences, null, 4);
 });
+
```
Comment (lacymorrow) [inline example/renderer.js:16]: We should use the contextBridge api (in `preload.js`)

**Comment by lacymorrow** [comment]
This is why PRs should be discrete, feature-specific code changes, there's just too much here to separate at this point. I've pulled your code for the button into it's own PR.

**Pull request closed without merging**